addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1603,54 +1612,21 @@ options.registry.anchorName = options.Class.extend({
     /**
      * @see this.selectClass for parameters
      */
-    openAnchorDialog: function (previewMode, value, $opt) {
-        var self = this;
-        var buttons = [{
-            text: _t("Save"),
-            classes: 'btn-primary',
-            click: function () {
-                var $input = this.$('.o_input_anchor_name');
-                var anchorName = $input.val().trim().replace(/\s/g, '_');
-                if (self.$target[0].id === anchorName) {
-                    // If the chosen anchor name is already the one used by the
-                    // element, close the dialog and do nothing else
-                    this.close();
-                    return;
-                }
-
-                var isValid = /^[\w-]+$/.test(anchorName);
-                var alreadyExists = isValid && $('#' + anchorName).length > 0;
-                var anchorOK = isValid && !alreadyExists;
-                this.$('.o_anchor_not_valid').toggleClass('d-none', isValid);
-                this.$('.o_anchor_already_exists').toggleClass('d-none', !alreadyExists);
-                $input.toggleClass('is-invalid', !anchorOK);
-                if (anchorOK) {
-                    self._setAnchorName(anchorName);
-                    this.close();
+    copyAnchor: function (previewMode, value, $opt) {
+        if (!this.$target[0].id) {
+            const titles = this.$target.find('h1, h2, h3, h4, h5, h6');
+            const title = titles.length > 0 ? titles[0] : this.$el.parent().find('we-title span')[0];
+            const anchorName = encodeURIComponent(title.innerText);
+            let n;
+            if (document.getElementById(anchorName)) {
+                n = 2;
+                while (document.getElementById(anchorName + n)) {
+                    n++;
```

Comment:
I think the code can be written to avoid the repetition of the condition with a do...while